src/main/cpp/JavaExceptionUtils.h

```diff
@@ -38,7 +38,7 @@ class JavaExceptionUtils {
 public:
   static EXPORT JniLocalRef<jobject> newThrowable(JNIEnv *env, const char *message, ...);
 
-  static EXPORT void checkException(JNIEnv *env);
+  static EXPORT void checkException(JNIEnv *env, bool clear = false);
```

Comment:
I'm not a huge fan of default arguments; it makes it difficult if we want to add another parameter to this method etc. Could we refactor to have 2 methods instead? I would propose calling it `checkExceptionAndClear()` and just skip the bool parameter.